dashboard/test/models/user_test.rb

```diff
@@ -2933,10 +2933,10 @@ def track_progress(user_id, script_level, result, pairings: nil)
   end
 
   test "age is set exactly for Google OAuth users between ages 4 and 20" do
-    four_year_old = create :user, birthday: (Date.today - 4.years), provider: 'google_oauth2'
+    four_year_old = build :user, birthday: (Date.today - 4.years), provider: 'google_oauth2'
```

Comment:
is it a problem that there is no `uid` here? apologies for my lack of deep understanding here. I am just asking because I see that it is specified by this trait: https://github.com/code-dot-org/code-dot-org/blob/71a185b0d554eac04ba27f2ec857708d856eec3b/dashboard/test/factories/factories.rb#L319-L323

Reply:
My understanding is that `user.uid` is the pre-authentication-options way of identifying oauth, and that that information has been migrated to `authentication_option.authentication_id`